code/game/objects/items/devices/laserpointer.dm

```diff
@@ -0,0 +1,218 @@
+/obj/item/device/laser_pointer
+	name = "laser pointer"
+	desc = "Don't shine it in your eyes!"
+	icon = 'icons/obj/device.dmi'
+	icon_state = "pointer"
+	item_state = "pen"
+	var/pointer_icon_state
+	slot_flags = SLOT_BELT
+	matter = list("glass" = 500,"metal" = 500)
+	w_class = 2 //Increased to 2, because diodes are w_class 2. Conservation of matter.
+	origin_tech = list(TECH_MAGNET = 2, TECH_COMBAT = 1)
+	var/turf/pointer_loc
+	var/energy = 8
+	var/max_energy = 8
+	var/effectchance = 20
+	var/cooldown = 10
+	var/last_used_time = 0
+	var/recharging = 0
+	var/recharge_locked = 0
+	var/obj/item/weapon/stock_parts/micro_laser/diode //used for upgrading!
+
+
+/obj/item/device/laser_pointer/red
+	pointer_icon_state = "red_laser"
+/obj/item/device/laser_pointer/green
+	pointer_icon_state = "green_laser"
+/obj/item/device/laser_pointer/blue
+	pointer_icon_state = "blue_laser"
+/obj/item/device/laser_pointer/purple
+	pointer_icon_state = "purple_laser"
+
+/obj/item/device/laser_pointer/New()
+	..()
+	diode = new(src)
+	if(!pointer_icon_state)
+		pointer_icon_state = pick("red_laser","green_laser","blue_laser","purple_laser")
+
+/obj/item/device/laser_pointer/upgraded/New()
+	..()
+	diode = new /obj/item/weapon/stock_parts/micro_laser/ultra
+
+
+
+/obj/item/device/laser_pointer/attack(mob/living/M, mob/user)
+	laser_act(M, user)
+
+/obj/item/device/laser_pointer/attackby(obj/item/W, mob/user)
+	if(istype(W, /obj/item/weapon/stock_parts/micro_laser))
+		if(!diode)
+			user.drop_item()
+			W.loc = src
+			diode = W
+			to_chat(user, "<span class='notice'>You install a [diode.name] in [src].</span>")
+		else
+			to_chat(user, "<span class='notice'>[src] already has a cell.</span>")
```

Comment:
`Cell` may be misleading here, as every other instance I can think of uses it in reference to a _power_ cell, as opposed to a _laser_ cell here.

Reply:
Didn't notice that, I'll change it.